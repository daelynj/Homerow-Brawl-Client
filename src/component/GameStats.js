import React from "react";

function GameStats(props) {
  return (
    <div>
      <div>{"words typed: " + props.wordsLength}</div>
      <div>
        {"WPM: " + Math.round(props.wordsLength / (props.finishTime / 60))}
      </div>
      <div>{"time: " + props.finishTime + " seconds"}</div>
    </div>
  );
}

export default GameStats;
