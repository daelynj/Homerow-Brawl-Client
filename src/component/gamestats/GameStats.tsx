import * as React from "react";

interface Props {
  words: String[];
  incorrectLetters: number;
  finishTime: number;
}

export const GameStats = (props: Props) => {
  const getLettersTyped = () => {
    return props.words.join("").length;
  };

  const calculateAccuracy = () => {
    let letters: number = getLettersTyped();
    let accuracy: any = (
      ((letters - props.incorrectLetters) / letters) *
      100
    ).toFixed(1);

    return accuracy > 0 ? accuracy : 0;
  };

  //https://www.speedtypingonline.com/typing-equations
  const calculateWPM = () => {
    let allTypedEntries: number = getLettersTyped();
    let averageWordLength: number = 5;
    let timePlayed: number = props.finishTime / 60;
    let grossWPM: number = allTypedEntries / averageWordLength / timePlayed;

    return Math.round(grossWPM);
  };

  return (
    <div>
      <div>{"words typed: " + props.words.length}</div>
      <div>{"WPM: " + calculateWPM()}</div>
      <div>{"time: " + props.finishTime + " seconds"}</div>
      <div>{"mistakes: " + props.incorrectLetters}</div>
      <div>{"accuracy: " + calculateAccuracy() + "%"}</div>
    </div>
  );
};
