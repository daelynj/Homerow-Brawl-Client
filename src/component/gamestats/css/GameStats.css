.stats {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  text-align: left;
  margin-left: auto;
  margin-right: auto;
  width: 72%;
  font-size: 14pt;
}
