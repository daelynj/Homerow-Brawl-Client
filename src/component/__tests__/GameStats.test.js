import React from "react";
import GameStats from "../GameStats";
import renderer from "react-test-renderer";

describe("GameStats", () => {
  it("renders the expected game stats", () => {
    const tree = renderer
      .create(<GameStats wordsLength={60} finishTime={60} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
