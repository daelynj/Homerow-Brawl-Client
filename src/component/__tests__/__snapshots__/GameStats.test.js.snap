// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStats renders the expected game stats 1`] = `
<div>
  <div>
    words typed: 60
  </div>
  <div>
    WPM: 60
  </div>
  <div>
    time: 60 seconds
  </div>
</div>
`;
